packages/website/docs/plugins.md

```diff
@@ -3,21 +3,266 @@ id: plugins
+import { createLocalStorageRecentSearchesPlugin } from '@algolia/autocomplete-plugin-recent-searches';
+
+const recentSearchesPlugin = createLocalStorageRecentSearchesPlugin({
+  key: 'navbar',
+});
+
+autocomplete({
+  // ...
+  openOnFocus: true,
+  plugins: [recentSearchesPlugin],
+});
+```
+
+If you want to store recent searches elsewhere, you can implement your own storage using [`createRecentSearchesPlugin`](createRecentSearchesPlugin).
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+import { createRecentSearchesPlugin } from '@algolia/autocomplete-plugin-recent-searches';
+
+const endpoint = 'https://example.com/recent-searches-storage';
+
+const recentSearchesPlugin = createRecentSearchesPlugin({
+  storage: {
+    getAll() {
+      return fetch(endpoint).then((response) => response.json());
+    },
+    onAdd(item) {
+      fetch(endpoint, { method: 'POST', body: JSON.stringify(item) });
+    },
+    onRemove(id) {
+      fetch(`${endpoint}/${id}`, { method: 'DELETE' });
+    },
+  },
+});
+
+autocomplete({
+  // ...
+  openOnFocus: true,
+  plugins: [recentSearchesPlugin],
+});
+```
+
+:::note
+
+The `getAll` function supports [promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), meaning you can plug it to any asynchronous API.
 
 :::
+
+#### Query Suggestions
+
+The [`query-suggestions`](createQuerySuggestionsPlugin) plugin allows you to plug [Algolia Query Suggestions](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/query-suggestions/js/) to your autocomplete.
+
+```js
+import algoliasearch from 'algoliasearch/lite';
+import { autocomplete } from '@algolia/autocomplete-js';
+import { createQuerySuggestionsPlugin } from '@algolia/autocomplete-plugin-query-suggestions';
+
+const searchClient = algoliasearch(
+  'latency',
+  '6be0576ff61c053d5f9a3225e2a90f76'
+);
+const querySuggestionsPlugin = createQuerySuggestionsPlugin({
+  searchClient,
+  indexName: 'instant_search_demo_query_suggestions',
+});
+
+autocomplete({
+  // ...
+  plugins: [querySuggestionsPlugin],
+});
+```
+
+#### Insights
+
+The [`algolia-insights`](createAlgoliaInsightsPlugin) plugin automatically sends click and conversion events to the [Algolia Insights API](https://www.algolia.com/doc/rest-api/insights/) whenever a user interacts with the autocomplete.
+
+
+```js
+import algoliasearch from 'algoliasearch/lite';
+import { autocomplete } from '@algolia/autocomplete-js';
+import { createAlgoliaInsightsPlugin } from '@algolia/autocomplete-plugin-algolia-insights';
+import insightsClient from 'search-insights';
+
+const appId = 'latency';
+const apiKey = '6be0576ff61c053d5f9a3225e2a90f76';
+const searchClient = algoliasearch(appId, apiKey);
+insightsClient('init', { appId, apiKey });
+
+const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({ insightsClient });
+
+autocomplete({
+  // ...
+  plugins: [algoliaInsightsPlugin],
+});
+```
+
+## Reference
```

Comment:
I've used the same types as documented in `createAutocomplete`. I think we can do a final pass to unify documented types once the API is finalized.